### ISSUE TITLE
Can't reach docker registry for pull through cache from earthly 

### ISSUE THREAD

**Issue by AlexMikhalev** [earthly/earthly#2741]
I can’t get the earthly pull-through cache working with earthly, 
I followed https://docs.earthly.dev/ci-integration/pull-through-cache, got the docker cache working, and as soon as I added config. 
```
mv ~/.earthly/config.yml.back ~/.earthly/config.yml  and run
earthly bootstrap && docker exec earthly-buildkitd cat /etc/buildkitd.toml 
```
earthly stops working or runs soo slooow - 2 minutes to output cached [github.com/earthly/hello-world+hello](http://github.com/earthly/hello-world+hello)
```
cat ~/.earthly/config.yml 
global:
  buildkit_additional_config: |
    [registry."docker.io"]
      mirrors = ["172.17.0.1:5000"]
    [registry."172.17.0.1:5000"]
      insecure = true
```
I tried multiple experiments with on and off earthly config and redirecting calls via /etc/hosts. It doesn’t look like earthly is hitting the docker registry cache. It seems internal build kit network routing is messed up - it can’t reach the docker gateway IP.
Following discussion in [Slack](https://earthlycommunity.slack.com/archives/C01DL2928RM/p1677406242893679) I tried to change insecure = true with http = true, but nothing changed.

**Comment by gfyrag** [comment]
Hi @AlexMikhalev !
I handle this on my side by starting a registry on a docker network, name "docker-mirrors".
Then connect the earthly buildkit container to this network using "docker connect".
And use this earthly config : 
```
global:
  buildkit_additional_config: |
    [registry."docker.io"]
      mirrors = ["registry-docker-io:5000"]
      insecure = true
      http = true
    [registry."registry-docker-io:5000"]
      http = true
```

**Comment by alexcb** [comment]
@AlexMikhalev did you solve this issue?

**Comment by AlexMikhalev** [comment]
nope, I will be re-doing the infrastructure network anyway, so I parked this for now.
